API_doc/Communication.md

```diff
@@ -4,14 +4,20 @@
 
 All requests must contain **REQUEST_NAME**, containing the type of action they want to do.
 Each response contain at least **REQUEST_STATE** (see below).
+All the local settings are applied after the config inclusions.
 
 ### Request types
 | Request Name | Request Description | Additional Argument(s) | Additional Returned Value(s) |
 | -------- | -------- | -------- | -------- |
 |*CONFIG_CREATE*|Create a new config.|**CONFIG_NAME**|**CONFIG_KEY**<br>**READONLY_CONFIG_KEY**|
 |*CONFIG_LOAD*| Load an existing config. |**CONFIG_KEY** *or* **READONLY_CONFIG_KEY**| **CONFIG_NAME**<br>**CONFIG_ID**|
 |*CONFIG_UNLOAD*| Unload config. |**CONFIG_ID**|*none*|
-|*CONFIG_INCLUDE*| Include a config in another |**CONFIG_ID**<br>**SRC**<br>**DEST**|*none*|
+|*CONFIG_DELETE*| Delete config. |**CONFIG_ID**|*none*|
+|*CONFIG_GET_DEPS*| Get the ordered list of all included configs. |**CONFIG_ID**|**DEPS** (array of READONLY_CONFIG_KEY)|
+|*CONFIG_GET_SETTINGS*| Get the list of all local settings. |**CONFIG_ID**|**SETTINGS** (map of settings : "SETTING_NAME" -> "SETTING_VALUE")|
+|*CONFIG_GET_ALIASES*| Get the list of all local aliases. |**CONFIG_ID**|**ALIASES** (map of aliases : "ALIAS_NAME" -> "SETTING_NAME")|
+|*CONFIG_INCLUDE*| Include a config |**CONFIG_ID**<br>**SRC** (READONLY_CONFIG_KEY of config to include)<br>**INCLUDE_POSITION** (position in the list of inclusion, where 0 is the first to be included. If not specified, is equal to *0*)|*none*|
+|*CONFIG_UNINCLUDE*| Uninclude a config |**CONFIG_ID**<br>**SRC** (READONLY_CONFIG_KEY of config to uninclude) *or* <br>**INDEX** (position in the list of inclusion)|*none*|
 |*SETTING_UPDATE*| Update or create setting |**CONFIG_ID**<br>**SETTING_NAME**<br>**SETTING_VALUE**|*none*|
 |*SETTING_REMOVE*| Remove setting |**CONFIG_ID**<br>**SETTING_NAME**|*none*|
 |*SETTING_GET*| Get setting |**CONFIG_ID**<br>**SETTING_NAME**|**SETTING_VALUE**|
```

Comment:
it just occured to me but i feel like we should be able to specify a list of settings to get the values from, rather than doing a request for each one at a time

Reply:
I agree with the idea. I'm not sure it's the best moment to do that however. I'll open an issue.